torch/csrc/jit/mobile/file_format.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <cstring>
+#include <fstream>
+#include <istream>
+
+/**
+ * @file
+ *
+ * Helpers for identifying file formats when reading serialized data.
+ *
+ * Note that these functions are declared inline because they will typically
+ * only be called from one or two locations per binary.
+ */
+
+namespace torch {
+namespace jit {
+
+/**
+ * The format of a file or data stream.
+ */
+enum class FileFormat {
+  UnknownFileFormat = 0,
+  FlatbufferFileFormat,
+  ZipFileFormat,
+};
+
+namespace internal {
+
+/// The size of the buffer to pass to #getFileFormat(), in bytes.
+constexpr size_t kFileFormatHeaderSize = 8;
+
+/**
+ * Returns the likely file format based on the magic header bytes in @p header,
+ * which should contain the first bytes of a file or data stream.
+ */
+// NOLINTNEXTLINE(facebook-hte-NamespaceScopedStaticDeclaration)
+static inline FileFormat getFileFormat(
+    const std::array<char, kFileFormatHeaderSize>& header) {
```

Comment:
Could we use c10/string_view here?

Reply:
I think it's better to deal with raw bytes because the data could contain `\0` elements, and that might break assumptions about string behavior.